File: chinese-shadowing-api/Shadowing.DataAccess/Entities/Series.cs
using System;
using System.Collections.Generic;

namespace Shadowing.DataAccess.Entities
{
    public class Series
    {
        public string Id { get; set; }
        public string Title { get; set; }
        public string UrlTitle { get; set; }
        public List<Episode> Episodes { get; set; }
        public DateTime Created { get; set; }
        public DateTime? Updated { get; set; }
    }
}

File: chinese-shadowing-client/src/models/Series.js
import Episode from './Episode';

export default class Series {
  /** @type {string} */
  id;
  /** @type {string} */
  title;
  /** @type {string} */
  urlTitle;
  /** @type {Episode[]} */
  episodes;

  constructor (series) {
    this.id = series.id;
    this.title = series.title;
    this.urlTitle = series.urlTitle;
    this.episodes = series.episodes.map(e => new Episode(e));
  }
}
